### DISCUSSION TITLE
Welcome to max30102-tutorial-raspberrypi Discussions!

### DISCUSSION THREAD

**Discussion by vrano714** [vrano714/max30102-tutorial-raspberrypi#9] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

## From the author of the repo

I don't think that this repo is a library to use MAX30102 on Raspberry Pi.
I just did re-write the Arduino sample codes to Python codes.

However, I understand some demand for providing APIs (like other MAX30xxx repos).
I hope we can do some discussion (use Ideas for suggesting ideas to enhance this repo).

General comments and tips are of course welcome.

**Comment by Abhishekcr448** [comment]
Hi, I was using your file hrcalc.py to find values of hr and spo2. And to check the code, I passed a demo code which will pass a few readings,(not 100). And I faced an error.
If you pass less number of readings to the hrcalc.py still it takes index values to 100, here in line no. 146👇
`while i < size - 1`
here if you are indexing till `size` which is 100 it results in IndexError If the list doesn't consist of those many values.
To solve that problem you just need to replace this: `while i < size - 1` with this: `while i < len(x) - 1:`
By this, the indexing will occur only till the last item of the list `x` and the code will run any no. of values of the list.

**Comment by Darkai87** [comment]
So when I hooked the MAX30102 sensor up properly to my Raspberry Pi Zero, I tried to run the program only to get the following errors in Thonny.

Line 5 in module: m = max30102.MAX30102()

Line 55 in __init__: self.bus = smbus.SMBus(self.channel)
FileNotFoundError: [Errno 2] No such file or directory 

Yet I have everything installed along with the python files in the same folder. If anyone could help with my situation I'd greatly appreciate it.
